docs/getting-started/index.md

```diff
@@ -0,0 +1,534 @@
+
+```
+starport serve
+```
+
+This will download dependencies, compile the source code into a binary called `blogd` (by default Starport uses the name of the repo + `d`), use this binary to initialize a single validator node, and start the node.
+
+![./images/api.png](./images/api.png)
+
+A validator node exposes two endpoints: [http://localhost:26657](http://localhost:26657) for the low-level Tendermint API and [http://localhost:1317](http://localhost:1317) for the high-level blockchain API.
+
+When you want to stop your blockchain, press Ctrl+C in the terminal window where it's running. In the development environment, Starport doesn't require you to restart the blockchain; changes you make in files will be automatically picked up by Starport.
+
+### Say "Hello, Starport"
+
+To get your Cosmos SDK blockchain to say "Hello", you will need to modify a protocol buffer file, create a keeper query function that returns data, and register a query function. Protocol buffer files contain proto `rpc`s that define Cosmos SDK queries and message handlers and proto `message`s that define Cosmos SDK types. `rpc`s are also responsible for exposing an HTTP API. [Keeper](https://docs.cosmos.network/v0.42/building-modules/keeper.html) is an abstraction for modifying the state of the blockchain and keeper functions let you query or write to the state. Registering a query needs to happen only once after you add the first query to your chain.
+
+In terms of workflow, developers usually work with proto files first to define Cosmos SDK messages, queries, message handlers and then implement the logic of these queries and message handlers in keeper functions.
+
+Let's start by creating a `posts` query:
+
+```
+starport query posts --response body,title
+```
+
+`query` accepts a name of the query (in our case, `posts`), an optional list of request parameters (in our case, empty), and an optional comma-separated list of response fields with a `--response` flag (in our case, `body,title`).
+
+The `query` command has created and modified several files:
+
+- modified `proto/blog/query.proto`
+- created `x/blog/keeper/grpc_query_posts.go`
+- modified `x/blog/client/cli/query.go`
+- created `x/blog/client/cli/query_posts.go`
+
+Let's examine some of these changes. For clarity, we'll skip placeholder comments Starport uses to scaffold code.
+
+In `proto/chain/query.proto` a `Posts` `rpc` has been added to the `Query` `service`.
+
+```
+service Query {
+	rpc Posts(QueryPostsRequest) returns (QueryPostsResponse) {
+		option (google.api.http).get = "/alice/blog/blog/posts";
+	}
+}
+```
+
+`Posts` `rpc` will be responsible for returning a list of all the posts on chain. It accepts request parameters (`QueryPostsRequest`) and returns response of type `QueryPostsResponse`. `option` defines the endpoint that will be used by gRPC to generate an HTTP API.
+
+Below you can see both request and response types.
+
+```
+message QueryPostsRequest {
+}
+
+message QueryPostsResponse {
```

Comment:
Not sure if relevant, but with the given command the entries are in reverse order

```
message QueryPostsResponse {
  string body = 1;
  string title = 2;
}
```